comparison_pipeline/compare.py

```diff
@@ -0,0 +1,103 @@
+"""Comparison file to compare scraped data with data in the db"""
+
+import pandas as pd
+from psycopg2.extensions import connection
+
+from extract import get_db_connection
+
+
+SCRAPED_DATA_TRANSFORMED = "transformed_data.csv"
+ARTICLE_CHANGES = "article_changes_deteceted.csv"
+
+
+def get_scraped_data_as_df(file_path: str) -> pd.DataFrame:
+    """Returns scraped data from extract"""
+
+    return pd.read_csv(file_path)
+
+
+def get_latest_version_of_article_from_db(conn: connection) -> pd.DataFrame:
+    """Returns data from rds database in a dataframe for comparison"""
+
+    cur = conn.cursor()
+
+    cur.execute("""SELECT 
+                article.article_id, article.article_url,
+                article_version.heading, article_version.body,
+                article_version.scraped_at,
+                string_agg(author.author_name, ',') AS author_name
+                FROM article
+                LEFT JOIN article_version
+                ON article_version.article_id = article.article_id
+                LEFT JOIN article_author
+                ON article_author.article_id = article.article_id
+                LEFT JOIN author
+                ON article_author.author_id = author.author_id
+                WHERE article_version.scraped_at = (SELECT MAX(article_version.scraped_at) 
+                FROM article_version WHERE article.article_id = article_version.article_id)
+                GROUP BY article.article_id, article.article_url, 
+                article_version.heading, article_version.body,
+                scraped_at
+                ;""")
+
+    data = cur.fetchall()
+
+    return pd.DataFrame(data, 
+                        columns=["article_id","article_url", "heading",
+                                 "body", "scraped_at", "author_name"])
+
+
+def split_author_name_row(row: str | None) -> list[str] | None:
```

Comment:
take out the '| None', '[str] | None' because it might break on older pythons

Reply:
We will dockerise this to pyhon:latest so this wont be a problem no?